### PR TITLE
bbrooks/3104-Preview-Add_Mongo_Server_NEW

### PR DESCRIPTION
**Pull request by cthulhuplus**
resolves # 3104

**Description-**
Adds a mongo server to the preview environment.

**This pull request changes...**

- Adds a mongo db to the preview environment

**This pull request also touches…**
CircleCI pipeline
aws user data file
aws deploy script

- possible side effects of this change
Pipeline or deployment is impacted

**This pull request was tested in the follow ways…**
Deployed to preview, logged in to MongoDB as root user and db user.

- list of frontend cases
- list of backend cases

**Steps to manually verify this change...**

1. SSH to the preview instance
2. Log in to the Mongo DB using the various users, do Mongo stuff

### This pull request is ready to review when...

- [ ] Automated tests are updated (and all tests are passing)
- [ ] The change has been documented
- [ ] Associated OpenAPI documentation has been updated
- [ ] Changelog is updated as appropriate
- [ ] The experience passes a basic manual accessibility audit (keyboard nav, screenreader, text scaling) OR an exemption is documented

### This pull request can be merged when…

- [ ] Code has been reviewed by someone other than the original author
- [ ] QA has verified the accessibility and functionality related to the change
- [ ] Design has approved the experience
- [ ] Product has approved the experience

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -661,9 +661,18 @@ jobs:
...
                 --JWT_SECRET "$PREVIEW_JWT_SECRET" \
+                --MONGO_INITDB_ROOT_USERNAME "$REVIEW_MONGO_INITDB_ROOT_USERNAME" \
+                --MONGO_INITDB_ROOT_PASSWORD "$PREVIEW_MONGO_INITDB_ROOT_PASSWORD" \
+                --MONGO_INITDB_DATABASE "$PREVIEW_MONGO_INITDB_DATABASE" \
+                --MONGO_DATABASE_USERNAME "$PREVIEW_MONGO_DATABASE_USERNAME" \
+                --MONGO_DATABASE_PASSWORD "$PREVIEW_MONGO_DATABASE_PASSWORD" \
                 --MONGO_DATABASE "$PREVIEW_MONGO_DATABASE" \
-                --MONGO_URL "$PREVIEW_MONGO_URL"
-
+                --MONGO_URL "$PREVIEW_MONGO_URL" \
```
Comment (thetif) [inline .circleci/config.yml:670]: duplicates?

File: docker-compose.yml [hunk 2]
```diff
@@ -1,62 +1,76 @@
...
+      - PGDATA=/pgdata
+    volumes:
+      - pgdata:/pgdata
+    expose:
+      - 5432
+    ports:
+      - 54325:5432
+
+  mongo:
+    image: mongo:4
```
Comment (thetif) [inline docker-compose.yml:17]: can you switch this to `mongo:latest`

**Review by thetif — CHANGES_REQUESTED** [review]
looks good, just two small things

File: .circleci/config.yml (hunk 1 above)
Comment (cthulhuplus) [reply .circleci/config.yml]: The duplicates should now be removed

File: docker-compose.yml (hunk 2 above)
Comment (cthulhuplus) [reply docker-compose.yml]: I have updated it to latest as requested

**Review by thetif — APPROVED** [review]
looks good, thanks!

**Pull request merged**